contracts/exchangeIssuance/FlashMintNAV.sol

```diff
@@ -0,0 +1,407 @@
+    * @param _inputToken            Address of the token used to pay for issuance. Use WETH address if input token is ETH.
+    * @param _inputTokenAmount      Exact amount of input token to spend
+    * @param _reserveAssetSwapData  Swap data to trade input token for reserve asset. Use empty swap data if input token is the reserve asset.
+    * 
+    * @return                       Amount of SetTokens expected to be issued
+    */
+    function getIssueAmount(
+        ISetToken _setToken,
+        address _inputToken,
+        uint256 _inputTokenAmount,
+        DEXAdapterV2.SwapData memory _reserveAssetSwapData
+    )
+        external
+        returns (uint256)
+    {
+        address reserveAsset = _getAndValidateReserveAsset(_setToken, _inputToken, _reserveAssetSwapData.path, true);
+        uint256 reserveAssetReceived = dexAdapter.getAmountOut(_reserveAssetSwapData, _inputTokenAmount);
+
+        return navIssuanceModule.getExpectedSetTokenIssueQuantity(
+            _setToken,
+            reserveAsset,
+            reserveAssetReceived
+        );
+    }
+
+    /**
+     * Gets the amount of output token expected to be received after redeeming a given quantity of Set Token
+     * for reserve asset and then swapping the reserve asset for output token.
+     * This function is not marked view, but should be static called from frontends.
+     * This constraint is due to the need to interact with the Uniswap V3 quoter contract
+     *
+    * @param _setToken              Address of the Set Token to be redeemed
+    * @param _setTokenAmount        Amount of Set Token to be redeemed
+    * @param _outputToken           Address of the token to send to caller after redemption
+    * @param _reserveAssetSwapData  Swap data to trade reserve asset for output token
+    * 
+    * @return                       Amount of output tokens expected to be sent to caller
+    */
+    function getRedeemAmountOut(
+        ISetToken _setToken,
+        uint256 _setTokenAmount,
+        address _outputToken,
+        DEXAdapterV2.SwapData memory _reserveAssetSwapData
+    )
+        external
+        returns (uint256)
+    {
+        address reserveAsset = _getAndValidateReserveAsset(_setToken, _outputToken, _reserveAssetSwapData.path, false);
+        uint256 reserveAssetReceived = navIssuanceModule.getExpectedReserveRedeemQuantity(
+            _setToken,
+            reserveAsset,
+            _setTokenAmount
+        );
+
+        return dexAdapter.getAmountOut(_reserveAssetSwapData, reserveAssetReceived);
+    }
+
+    /**
+    * Issues a minimum amount of SetTokens for an exact amount of ETH.
+    *
+    * @param _setToken              Address of the SetToken to be issued
+    * @param _minSetTokenAmount     Minimum amount of SetTokens to be issued
+    * @param _reserveAssetSwapData  Swap data to trade WETH for reserve asset
+    */
+    function issueSetFromExactETH(
+        ISetToken _setToken,
+        uint256 _minSetTokenAmount,
```

Comment:
@janndriessen double-checking that you're able to integrate with this issuance function that doesn't return an exact amount of SetTokens?

Reply:
@edkim so we'd have to call `getIssueAmount` or `getRedeemAmountOut` to determine this `minSetTokenAmount`?

Reply:
@janndriessen exactly. See example for issuance here https://github.com/IndexCoop/index-coop-smart-contracts/pull/186/files#r1768661768.

And for redemption https://github.com/IndexCoop/index-coop-smart-contracts/pull/186/files#r1768665399